lib/genre.rb

```diff
@@ -0,0 +1,35 @@
+class Genre
+  extend Concerns::Findable
+
+  @@all = []
+
+  def initialize(name)
+    @name = name
+    @songs = []
+  end
+
+  attr_accessor :name
+  attr_accessor :songs
+
+  def self.all
+    @@all
+  end
+
+  def save
+    @@all << self
+    self
+  end
+
+  def self.destroy_all
+    @@all.clear
+  end
+
+  def self.create(name)
+    @@all.detect { |genre| genre.name == name} ||
```

Comment:
Consider having this method do only one thing: *create an artist*. Leave the rest to the method *find_or_create_by_name*.